client/js/screens/MainPage/MainPage.js

```diff
@@ -1,9 +1,19 @@
 import React from "react";
-import { Text } from "react-native";
+import { Text, View } from "react-native";
+import {withNavigation} from 'react-navigation'
 import styles from "./styles";
 
-const MainPage = () => {
-  return <Text>I am the MainPage Screen</Text>;
+const MainPage = ({ navigation }) => {
+  return(  <View>
+    <Text>I am the MainPage Screen</Text>
+    <Text
+      onPress={() => {
+        navigation.navigate("WorkoutSession");
+      }}
+    >
+      To WorkoutSession
+    </Text>
+  </View>)
 };
 
-export default MainPage;
+export default withNavigation(MainPage);
```

Comment:
Remove withNavigation. You do not need it. Use your container to pass navigation methods and properties as props for this component.